src/plone/restapi/services/users/get.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+from plone import api
+from plone.restapi.services import Service
+from zope.interface import implements
+from zope.publisher.interfaces import IPublishTraverse
+
+
+class UsersGet(Service):
+
+    implements(IPublishTraverse)
+
+    def __init__(self, context, request):
+        super(UsersGet, self).__init__(context, request)
+        self.params = []
+
+    def publishTraverse(self, request, name):
+        # Consume any path segments after /@users as parameters
+        self.params.append(name)
+        return self
+
+    @property
+    def _get_user_id(self):
+        if len(self.params) != 1:
+            raise Exception(
+                "Must supply exactly one parameter (user id)")
+        return self.params[0]
+
+    def reply(self):
+        if len(self.params) == 0:
+            return [
+                {
+                    'id': user.id,
+                    'email': user.getProperty('email'),
+                    'username': user.getUserName(),
+                    'fullname': user.getProperty('fullname'),
+                    'home_page': user.getProperty('home_page'),
+                    'description': user.getProperty('description'),
+                    'location': user.getProperty('location')
+                } for user in api.user.get_users()
```

Comment:
There's a caveat here - Zope's `Products.PluggableAuthenticationService` does not support an interface that *lists* all users, it only provides an interface [`IUserEnumerationPlugin`](https://github.com/zopefoundation/Products.PluggableAuthService/blob/master/Products/PluggableAuthService/interfaces/plugins.py#L292) that can be used to *query / search* users.
`PlonePAS` on the other hand introduces [`IUserIntrospection`](https://github.com/plone/Products.PlonePAS/blob/master/src/Products/PlonePAS/interfaces/plugins.py#L6) which allows for a complete listing, and which is what `plone.api.user.get_users()` uses.

Unfortunately, `Products.LDAPUserFolder` does not implement an `IUserIntrospection` plugin, which is why on a site with all your users in LDAP, `api.user.get_users()` will return an empty list.

From what I understood when I looked into this surprising (to me) behavior a while ago, this was a deliberate decision. Enumerating all users can, depending on the backend and amount of users, absolutely kill performance, and that's why it was considered "an API that shouldn't be used, and therefore isn't provided" by the authors of `Products.PluggableAuthService` (IIRC). There's a rather heated discussion on the details of this [here (dzug ML, german)](http://mail.dzug.org/mailman/archives/zope/2006-December/002431.html).

I still think exposing the behavior of `plone.api` is the right thing to do, but we might want to point out this caveat in the documentation.